src/Microsoft.Health.Fhir.SqlServer/Features/ChangeFeed/SqlServerFhirResourceChangeDataStore.cs

```diff
@@ -0,0 +1,111 @@
+
+        // dbnetlib error value for timeout expired
+        public const short TIMEOUTEXPIRED = -2;
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="SqlServerFhirResourceChangeDataStore"/> class.
+        /// </summary>
+        /// <param name="sqlConnectionFactory">The SQL Connection factory.</param>
+        /// <param name="logger">The logger.</param>
+        public SqlServerFhirResourceChangeDataStore(ISqlConnectionFactory sqlConnectionFactory, ILogger<SqlServerFhirResourceChangeDataStore> logger)
+        {
+            EnsureArg.IsNotNull(sqlConnectionFactory, nameof(sqlConnectionFactory));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _sqlConnectionFactory = sqlConnectionFactory;
+            _logger = logger;
+        }
+
+        /// <summary>
+        ///  Returns the number of resource change records from startId.
+        /// </summary>
+        /// <param name="startId">The start id of resource change records to fetch. The start id is inclusive.</param>
+        /// <param name="pageSize">The page size for fetching resource change records.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>Resource change data rows.</returns>
+        /// <exception cref="System.ArgumentOutOfRangeException">Thrown if startId or pageSize is less than zero.</exception>
+        /// <exception cref="System.InvalidOperationException">Thrown when a method call is invalid for the object's current state.</exception>
+        /// <exception cref="Microsoft.Data.SqlClient.SqlException">Thrown when SQL Server returns a warning or error.</exception>
+        /// <exception cref="System.TimeoutException">Thrown when the time allotted for a process or operation has expired.</exception>
+        /// <exception cref="System.Exception">Thrown when errors occur during execution.</exception>
+        public async Task<IReadOnlyCollection<IResourceChangeData>> GetRecordsAsync(long startId, int pageSize, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsGte(startId, 0, nameof(startId));
+            EnsureArg.IsGte(pageSize, 0, nameof(pageSize));
+
+            var listResourceChangeData = new List<ResourceChangeData>();
+            try
+            {
+                using (SqlConnection sqlConnection = await _sqlConnectionFactory.GetSqlConnectionAsync(cancellationToken: cancellationToken))
+                {
+                    await sqlConnection.OpenAsync(cancellationToken);
+                    using (SqlCommand sqlCommand = new SqlCommand("dbo.FetchResourceChanges", sqlConnection))
+                    {
+                        sqlCommand.CommandType = CommandType.StoredProcedure;
+                        sqlCommand.Parameters.AddWithValue("@startId", SqlDbType.BigInt).Value = startId;
+                        sqlCommand.Parameters.AddWithValue("@pageSize", SqlDbType.Int).Value = pageSize;
```

Comment:
>SqlDbType.Int [](http://example.com/codeflow?start=72&length=13)

I think you changed it to smallint in other PR.

Reply:
Changed the data type of page size to smallint.